src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
```

Comment:
The class implements the Strategy pattern correctly by using a map to associate different operations with their respective handlers. This avoids the use of switch-case or if-else constructions for determining the operation handler, which is in line with the checklist's recommendations.